### PR TITLE
fix: draggable Modal should have boundary

### LINKED ISSUE #28438: [Modal] 自定义渲染对话框 Draggable 文档应当优化为只在 body 内可拖拽
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://stackblitz.com/edit/react-mp1hmz?file=index.js](https://stackblitz.com/edit/react-mp1hmz?file=index.js)

### Steps to reproduce
step1: 打开可拖拽 modal

step2: 拖拽至屏幕外

step3: 关闭 modal 再打开 modal

会发现 modal 再也拖不回来了

### What is expected?
draggable 在 body 内，需要优化下这处文档。

### What is actually happening?
modal 可以被拖拽至视口外，并且再也无法被拖拽回来

| Environment | Info |
|---|---|
| antd | 4.9.4 |
| React | ^16.9.44 |
| System | Mac OS |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by mumiao**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
fix #28438

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English   |  -  |
| 🇨🇳 Chinese |  - |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/modal/demo/modal-render.md](https://github.com/mumiao/ant-design/blob/fix/dialog/components/modal/demo/modal-render.md)

### CODE REVIEW

**Comment by mumiao** [comment]
@yoyo837 那个pr分支混杂了别的commits 我重新提到这里来

File: components/modal/demo/modal-render.md [hunk 1]
```diff
@@ -43,7 +51,21 @@ class App extends React.Component {
     });
   };
 
+  setBounds = () => {
+    const { clientWidth, clientHeight } = window?.document?.documentElement;
+    const targetRect = this.draggleRef?.current.getBoundingClientRect();
+    this.setState({
+      bounds: {
+        left: -targetRect.left,
+        right: clientWidth - targetRect?.right,
```
Comment (yoyo837) [inline components/modal/demo/modal-render.md:60]: 是不是应该处理一些NaN?

**Comment by yoyo837** [comment]
窗口resize后，依然可以拖拽到外面

**Comment by mumiao** [comment]
> 窗口resize后，依然可以拖拽到外面

done

File: components/modal/demo/modal-render.md (hunk 1 above)
Comment (mumiao) [reply components/modal/demo/modal-render.md]: 不存在为非number的情况把？

**Comment by yoyo837** [comment]
@xrkffgg 这个CI failed和 改动的action 有关系不？

**Comment by xrkffgg** [comment]
> @xrkffgg 这个CI failed和 改动的action 有关系不？

没关系

**Comment by mumiao** [comment]
@yoyo837 借助react-draggable实现限制边界，我尝试了几次；目前效果ok；review下代码，有问题我会及时跟进

**Comment by mumiao** [comment]
@afc163 @yoyo837 CI 报错跟我此次提交没有关系。 review一下看看？

**Pull request merged**